Hs 4052

palegre-fluendo opened the pull request:
Added adaptive-playback property to AMC video decoders.

aslobodeniuk commented on sys/androidmedia/gstamcaudiodec.c:
```diff
@@ -1139,7 +1139,8 @@ gst_amc_audio_dec_set_format (GstAudioDecoder * decoder, GstCaps * caps)
 
-  if (!gst_amc_codec_configure (self->codec, format, NULL, self->drm_ctx, 0, 0)) {
+  if (!gst_amc_codec_configure (self->codec, format, NULL, self->drm_ctx, 0, 0,
+          GST_AMC_DEFAULT_ADAPTIVE_PLAYBACK_ENABLED)) {
```
for audio codec adaptive playback doesn't exist, so it should be always FALSE

aslobodeniuk commented on sys/androidmedia/gstamc.h:
```diff
@@ -72,6 +72,7 @@ struct _GstAmcCodec
   guint flush_id;
   GMutex buffers_lock;
   gboolean tunneled_playback_enabled;
+  gboolean enable_adaptive_playback;
```
this field doesn't seem to be used anywhere. We initialize it to TRUE in gst_amc_codec_new , but never use it.

palegre-fluendo committed:
Add support for enable/disable adaptive-playback

New boolean property "adaptive-playback"  to enable/disable adaptive-playback, if available, on android platforms. By default it is enabled.
